### DISCUSSION TITLE
Allow for more flexible configuration of EventLog file path

### DISCUSSION THREAD

**Discussion by BartChris** [pkp/pkp-lib#9886] [Proposals]
We are currently using OJS 3.3 and storing our assets using an NFS mount. Using this setup, we encountered an issue where our journals became inaccessible for prolonged periods. We suspect that we are encountering the same problem described in this forum thread:: 
https://forum.pkp.sfu.ca/t/ojs-page-stalls-awesome-delays/4522/8

>  During high traffic a lot of apache processes want to write their logs parallel into the usageEventLogs - and the NFS mount is not handling the file-locks very quick. Therefore the processes (users) had to wait in the queue to get write-permissions leading to the awesome delay times.

We've noticed significant changes in the handling of statistics in OJS 3.4 (as documented here:  (https://github.com/pkp/pkp-lib/issues/6782), but we're unsure if these changes might also address our problem with storing the eventLog in the NFS.

Therefore, I would like to inquire whether it's possible to enable a more flexible configuration of the EventLog Path in OJS 3.4. Currently, it appears that OJS 3.4 still relies on the configured BasePath for the EventLog (as seen here: https://github.com/pkp/pkp-lib/blob/7cbab212a90a3a879e74333a672d4b1f72d7ad04/classes/statistics/PKPStatisticsHelper.php#L88).  Would it be feasible to extend the configuration options to allow specifying a different path for the EventLog?

**Comment by asmecher** [comment]
@BartChris, I'd be open to reviewing a pull request proposal for this (and will ask for @bozana's opinion here too), but it wouldn't be a high priority for the dev team to make the changes.
